src/main/java/com/fasterxml/jackson/databind/introspect/JacksonAnnotationIntrospector.java

```diff
@@ -240,6 +240,45 @@ public String[] findEnumValues(Class<?> enumType, Enum<?>[] enumValues, String[]
         return names;
     }
 
+    @Override
+    public String[] findEnumValues(MapperConfig<?> config, Enum<?>[] enumValues, String[] names,
+                                   AnnotatedClass annotatedClass) {
+        // First collect all JsonProperty.value()
+        HashMap<String, String> enumToPropertyMap = new HashMap<>();
+        for (AnnotatedField field : annotatedClass.fields()) {
+            JsonProperty property = field.getAnnotation(JsonProperty.class);
+            if (property != null) {
+                enumToPropertyMap.put(field.getName(), property.value());
+            }
+        }
+        final Class<?> enumType = annotatedClass.getRawType();
+        HashMap<String,String> expl = null;
+        for (Field f : enumType.getDeclaredFields()) {
+            if (!f.isEnumConstant()) {
+                continue;
+            }
+            String n = enumToPropertyMap.get(f.getName());
+            if (n == null || n.isEmpty()) {
```

Comment:
Ah ok, so empty String is handled here. Could have just not added it in Map earlier.

Reply:
Done 👍🏻 Revised(moved) this check to right before adding it to the Map.